### DISCUSSION TITLE
Generating text content for Text-To-Speech (TTS) software

### DISCUSSION THREAD

**Discussion by balacij** [JacquesCarette/Drasil#2987] [Ideas]
At times, I'd like to have a TTS read some block of text for me, mainly with PDFs. However, PDF to TTS seems to be problematic, they usually rush around certain areas, speak awkwardly, speak in odd tones, get confused with certain areas (for example, figures [an important one, in my opinion] and tables are nearly impossible to read), and section separation is confusing to the listener at times. Some PDFs are better than others for reading, it seems as though they were potentially made a bit more friendly to TTS software (I'm not sure if it was deliberate or not). We have extra information around the _real content_ of a potentially inputted PDF into a TTS software. Using this extra information, I think we can generate [Speech Synthesis Markup Language](https://en.wikipedia.org/wiki/Speech_Synthesis_Markup_Language) (or another language) from a TTS-variant of the SRS template (or anything else we may so choose) that has the listener in mind specifically, instead of a reader.

Although this is only another instance of language -> language transformation, rapidly improving accessibility, specifically, could be one of the extra selling points of Drasil/Drasils ideology too.

Are there any thoughts on this? It may not be worth implementing (at this time, or at all), but it might be something to remember for other contexts.

**Comment by JacquesCarette** [comment]
Fantastic use of 'Discussion'! Great idea - we've never discussed it. It is intriguing, as it is neither very far away, nor particularly close, kind of in that grey zone of how much work it would be to do.

We'd need to upgrade our capture of English a fair amount. But if we did, this ought to become straightforward. Very interesting idea indeed.

**Comment by smiths** [comment]
I agree with @JacquesCarette on all points.  Nice to see the Discussion board used for something and a very intriguing idea.  With a little homework on where to target this idea, I feel like it would also have the potential to attract funding.  Improving accessibility is a great goal - both on its own merits and because it is timely.

I'm assuming @balacij that this is a potential idea for your PhD.  Are you keeping your PhD ideas (and big picture ideas) together in one spot?  Any ideas you don't use, we can move to our [potential projects wiki](https://github.com/JacquesCarette/Drasil/wiki/Potential-Projects).

**Comment by balacij** [comment]
Thank you both! I'm glad the idea fits in the "Discussions" area nicely.

@JacquesCarette: Regarding the capture of English required that you mentioned, did you have any specific/immediate thoughts on what needed to be upgraded?

I never really thought this could be a PhD topic, but thinking about it now, it might be a good one. I'll try to think about it more and see if I can come up with anything. I will write something here if I can come up with anything. For now, I've written out my PhD topics on a notepad, but I'll be posting them once I've gone a bit further on my MSc thesis rough draft.

**Reply by JacquesCarette** [reply]
It's a dangerous topic, in that we have little expertise in the team, and the department, in the domain. 

As to what needs done: I think we need a fairly radical overhaul of the language capture that would be decidedly semantics-based instead of being syntax-based the way we are now. We'd probably need to adopt something like Grammatical Framework as our basis. That ought to work for the internals, but then rebuilding a decent user experience on top of that is something I have no clue how to do. This would all need a lot of investigation and experimentation (outside Drasil itself) to figure out.

**Reply by balacij** [reply]
Thank you! The English-part indeed sounds like it would be a huge problem on its own. In that case, I wonder if we could work on considerably smaller proof of concept problems, not necessarily related to the SRS or code, that are similar.

**Comment by smiths** [comment]
@balacij, you definitely don't have to prioritize putting your PhD topic ideas in electronic form, but I selfishly would like you to do when you have time.  It is always good to capture good ideas.  :smile:

**Reply by balacij** [reply]
I will try my best to get it written electronically! :smile: I will write them out with the same Purpose, Problem Statement, and Questions (PPSQ) format that I used for my thesis.
